platform.md

```diff
@@ -276,70 +276,17 @@ All lifecycle phases:
 - MUST give command line inputs precedence over other inputs
 
-#### `detector`
-The platform MUST execute `detector` in the **build environment**
-
-Usage: 
-```
-/cnb/lifecycle/detector \
-  [-app <app>] \
-  [-buildpacks <buildpacks>] \
-  [-group <group>] \
-  [-layers <layers>] \
-  [-log-level <log-level>] \
-  [-order <order>] \
-  [-plan <plan>] \
-  [-platform <platform>]
-```
-
-##### Inputs
-| Input         | Environment Variable    | Default Value             | Description
-|---------------|-------------------------|---------------------------|----------------------
-| `<app>`         | `CNB_APP_DIR`         | `/workspace`          | Path to application directory
-| `<buildpacks>`  | `CNB_BUILDPACKS_DIR`  | `/cnb/buildpacks`     | Path to buildpacks directory (see [Buildpacks Directory Layout](#buildpacks-directory-layout))
-| `<group>`       | `CNB_GROUP_PATH`      | `<layers>/group.toml` | Path to output group definition
-| `<layers>`      | `CNB_LAYERS_DIR`      | `/layers`             | Path to layers directory
-| `<log-level>`   | `CNB_LOG_LEVEL`       | `info`                | Log Level
-| `<order>`       | `CNB_ORDER_PATH`      | `/cnb/order.toml`     | Path to order definition (see [`order.toml`](#ordertoml-toml))
-| `<plan>`        | `CNB_PLAN_PATH`       | `<layers>/plan.toml`  | Path to output resolved build plan
-| `<platform>`    | `CNB_PLATFORM_DIR`    | `/platform`           | Path to platform directory
-
-##### Outputs
-| Output             | Description
-|--------------------|----------------------------------------------
-| [exit status]      | (see Exit Code table below for values)
-| `/dev/stdout`      | Logs (info)
-| `/dev/stderr`      | Logs (warnings, errors)
-| `<group>`          | Detected buildpack group  (see [`group.toml`](#grouptoml-toml))
-| `<plan>`           | Resolved Build Plan (see [`plan.toml`](#plantoml-toml))
-
-| Exit Code | Result|
-|-----------|-------|
-| `0`       | Success
-| `11`      | Platform API incompatibility error
-| `12`      | Buildpack API incompatibility error
-| `1-10`, `13-99` | Generic lifecycle errors
-| `100`     | All buildpacks groups have failed to detect w/o error
-| `101`     | All buildpack groups have failed to detect and at least one buildpack has errored
-| `102-199` | Detection-specific lifecycle errors
-
-The lifecycle:
-- SHALL detect a single group from `<order>` and write it to `<group>` using the [detection process](buildpack.md#phase-1-detection) outlined in the Buildpack Interface Specification
-- SHALL write the resolved build plan from the detected group to `<plan>`
-
 #### `analyzer`
 Usage: 
 ```
 /cnb/lifecycle/analyzer \
   [-analyzed <analyzed>] \
-  [-cache-dir <cache-dir>] \
   [-cache-image <cache-image>] \
   [-daemon] \ # sets <daemon>
   [-gid <gid>] \
   [-group <group>] \
   [-layers <layers>] \
   [-log-level <log-level>] \
-  [-skip-layers <skip-layers>] \
   [-uid <uid>] \
   <image>
```

Comment:
After looking over the `creator` usage and thinking about a future where we might want to validate write access to the destination tags in `analyzer` I think we should replace the `<image>` positional argument with an optional `-previous-image` flag. If we later introduce a positional argument to designate the destination tag we can default `previous-image` to image like we do in `creator`.

We typically use the positional arguments for required configuration. The previous image was formerly required by analyzer, but I think that is unintuitive on a first build as we should take the opportunity to change it while we are changing the usage here.

Should we do this now?:
Pros: It will be easier for platform implementers if we bundle more changes together
Cons: It will delay the next lifecycle release

Reply:
I'm fine with making this change now. I'll update the spec

Reply:
Hmm - I guess this would look like `-previous-image` needing to be required for Platform API < 0.6?

Reply:
I got this working in the PR. Commit here https://github.com/buildpacks/lifecycle/pull/505/commits/5988b8a58c818d4d4a6032d78b931fcc42548cdc

It could probably use some 👀  and potential spec updates to reflect the new behavior. Like when you don't have a `-previous-image`, do we need to explain which parts will and will not occur? I didn't write any tests for that yet. For `0.6` _right now_ omitting `-previous-image` and would result in an empty `analyzed.toml` I think. No metadata or image id would exist in that case. Obviously we want to start storing things about the run-image including the stack/mixins so detector and other phases could consume that information analzyed in this first step.

Reply:
oh, just kidding. I'll need to make sure in 0.5 and below the positional argument is allowed/required like it is today and not allow `-previous-image`. I'll fix that up tomorrow hopefully